Project 1/Part_2_Pump_Spec.lts

```diff
@@ -1,151 +1,210 @@
+// - In case of low battery, we call low_battery (example of Non-insistent alarm)°
+// - if the alarm system fails, we call alarm_system_fail (example of continuous alarm)
+//
+SETUP[params:ParamsStateT][lineLock:LineLockStateT][power:CordConnectedStateT]  = 
+(
+    unplug -> UNPLUGGED
+    |
+    turn_off -> POWER_OFF
+    |
+    when (params == ParamsNotSet && lineLock == LineUnlocked)
+        set_rate -> enter_value ->
+            (press_set -> SETUP[ParamsSet][lineLock][power]
+             |
+             press_cancel -> SETUP[ParamsNotSet][lineLock][power])
+    |
+    when (params == ParamsSet && lineLock == LineUnlocked)
+        clear_rate -> SETUP[ParamsNotSet][lineLock][power]
+    |
+    when (params == ParamsSet && lineLock == LineUnlocked)
+        connect_set -> purge_air -> lock_line -> SETUP[params][LineLocked][power]
+    |
+    when (lineLock == LineLocked)
+        confirm_settings -> INFUSION[UnitUnlocked][power]
+    |
+    when (lineLock == LineLocked)
+        erase_and_unlock_line -> SETUP[params][LineUnlocked][power]
+    |
+    when (power == CordConnected)                             // If the cord is connected and the AC power fails, switch to the battery
+        ac_power_fail -> AC_BATTERY_POWER[CordNotConnected]
+    |
+    when (power == CordNotConnected)                          // If the pump is using the battery, we can switch to the AC power
+        recover_ac_power -> AC_BATTERY_POWER[CordConnected]
+    |
+    low_battery -> NON_INSISTENT_ALARM                        // at any time, we can have a non insistent alarm
+    |
+    alarm_system_fail -> CONTINUOUS_ALARM                     // at any time, we can have a continuous alarm
+),
+
+//
+// Pump in infusion mode:
+// - Always be able to turn the unit off, even if locked
+// - Allow the user to lock/unlock the unit
+// - Errors could occur with the pump (e.g., line became pinched or plugged)
+//
+INFUSION[unitLock:UnitLockStateT][power:CordConnectedStateT] =
+(
+    unplug -> UNPLUGGED
+    |
+    turn_off -> POWER_OFF
+    |
+    when (unitLock == UnitUnlocked)
+        change_settings -> SETUP[ParamsSet][LineLocked][power]
+    |
+    when (unitLock == UnitUnlocked)
+        lock_unit -> INFUSION[UnitLocked][power]
+    |
+    when (unitLock == UnitLocked)
+        unlock_unit -> INFUSION[UnitUnlocked][power]
+    |
+    dispense_main_med_flow -> INFUSION[unitLock][power]
+    |
+    flow_blocked -> sound_alarm[1] -> INFUSION_BLOCKED[unitLock][1][power]
+    |
+    low_battery -> NON_INSISTENT_ALARM                           // at any time, we can have a non insistent alarm
+    |
+    alarm_system_fail -> CONTINUOUS_ALARM                        // at any time, we can have a continuous alarm
+),
+
+//
+// Pump stuck (medication flow suspended)
+//
+INFUSION_BLOCKED[unitLock:UnitLockStateT][alarm:AlarmStateT][power:CordConnectedStateT] =
+(
+    unplug -> UNPLUGGED
+    |
+    turn_off -> POWER_OFF
+    |
+    when (alarm == 1)
```

Comment:
Instead of using `alarm == 1` kind of comparison, can we maintain the `alarm == AlarmActivated` pattern.